tools/engine_tool/lib/src/build_plan.dart

```diff
@@ -84,14 +90,48 @@ final class BuildPlan {
     required this.useRbe,
     required this.useLto,
     required this.concurrency,
-  }) {
+    required Iterable<String> extraGnArgs,
+  }) : extaGnArgs = List.unmodifiable(extraGnArgs) {
     if (!useRbe && strategy == BuildStrategy.remote) {
       throw FatalError(
         'Cannot use remote builds without RBE enabled.\n\n$_rbeInstructions',
       );
     }
   }
 
+  /// Arguments that cannot be provided to [BuildPlan.extraGnArgs].
+  ///
+  /// Instead, provide them explicitly as other [BuildPlan] arguments.
+  @visibleForTesting
+  static const reservedGnArgs = {
+    _flagRbe,
+    _flagLto,
+    'no-$_flagRbe',
+    'no-$_flagLto',
+  };
+
+  static void _checkExtraGnArgs(Iterable<String> gnArgs) {
+    for (final arg in gnArgs) {
+      if (!arg.startsWith('--') || arg.contains('=') || arg.contains(' ')) {
+        throw FatalError(
+          'Arguments provided to --gn-args must be flags (booleans) and be '
+          'specified as either in the format "--flag". Options that are not '
+          'flags or are abberviated ("-F") are not currently supported: '
+          'https://fluter.dev/to/engine-tool-bug.',
+        );
+      }
+
+      // Strip off the prefix and compare it to reserved flags.
+      final withoutPrefix = arg.replaceFirst('--', '');
+      if (reservedGnArgs.contains(withoutPrefix)) {
+        throw FatalError(
+          'Both --rbe and --lto (and their --no-x variants) should be '
+          'specified as direct arguments to "et" and not using "--gn-args".',
```

Comment:
nit: the error message should probably be generated from reservedGnArgs

Reply:
Done!